### DISCUSSION TITLE
Slash Command에 대한 대책

### DISCUSSION THREAD

**Discussion by wonderlandpark** [koreanbots/koreanbots#140] [일반] [answered]
# Slash Command

[API Docs](https://github.com/discord/discord-api-docs/blob/a4e5d78d4dfada4a77cbaff4c5a3a40797a641e2/docs/interactions/Slash_Commands.md)

최근 디스코드에 **Slash Command**라는 기능이 추가되었습니다. 이를 사이트에서 어떻게 표현하면 좋을지, 의견을 받고자합니다.

**Comment by wonderlandpark** [comment]
일단 저는 봇을 등록할 때 Slash Command 지원 유/무를 선택하도록 하고, 수정할 수도 있게할 생각입니다. KOREANBOTS에서도 `applications.commands` 스코프를 Slash Command를 지원하는 봇으로 한정하여 부여할 생각이지만 많은 봇이 표시되어 다소 난잡할 수 있지 않을까 고민됩니다.

**Comment by wonderlandpark** [comment]
또한 사이트에서 Slash Command를 지원하는 봇에게는 별도로 UI에서 표시를 하고자하는데 어떻게 하면 좋을까요?

**Reply by wonderlandpark** [reply]
그냥 카테고리를 추가할까요?

**Reply by zero734kr** [reply]
카테고리보다는 라벨 같은 뱃지가 더 낫지 않나 싶습니다 카테고리는 더 난잡해질것 같아서

**Comment by kiki7000** [comment]
서버에선 그냥 다른 채널에는 안되게 하고(가능하면) 새 채널을 만들어 거기서만 되게 하는거 어때요?

**Reply by wonderlandpark** [reply]
그 생각을 못했네요 서버에서 특정 채널만 금지는 안될 수도 있겠네요

**Reply by kiki7000** [reply]
현재 슬래시 커맨드 권한이 나왔으니까 이렇게 하는것 어떤가요?

**Comment by crafthome0** [comment]


**Reply by crafthome0** [reply]
아 위에 의견이 있네요

**Comment by wonderlandpark** [comment]
슬래시 커맨드의 채널별 권한 설정이 나오기 전까지는 서버에서 권한을 부여하는건 무리일듯합니다.

하지만 사이트에는 slash command여부를 표시하는 방향으로 진행해보겠습니다.

**Comment by wonderlandpark** [comment, marked as answer]
디스코드에 채널별 빗금 명령어 사용 권한을 제한할 수 있어서 다음과 같이 하도록 하겠습니다.

1. 사이트내에서는 디스코드가 채택한 번역인 `빗금 명령어`를 사용해 카테고리 추가
2. 한국 디스코드봇 리스트 디스코드에 추가할 때 빗금 명령어 권한 부여를 하지만, 테스트 채널에서만 사용할 수 있도록 제한
